src/circuit/qc_keyagg.rs

```diff
@@ -187,85 +365,180 @@ mod tests {
-        assert!(circuit.check_circuit_satisfiability(&[]).is_ok());
+        circuit.check_threshold(&stake_amt_vars[..], &selector_vars[..], threshold_var)?;
+        assert!(circuit.check_circuit_satisfiability(&public_input).is_ok());
 
         // bad path: wrong aggregated vk
-        let tmp_var = agg_vk_var.0 .0.to_vec()[0];
+        let tmp_var = agg_vk_var.native_vars()[0];
         let tmp = circuit.witness(tmp_var)?;
         *circuit.witness_mut(tmp_var) = F::zero();
-        assert!(circuit.check_circuit_satisfiability(&[]).is_err());
+        assert!(circuit.check_circuit_satisfiability(&public_input).is_err());
         *circuit.witness_mut(tmp_var) = tmp;
 
         // bad path: wrong digest
         let tmp = circuit.witness(digest_var)?;
         *circuit.witness_mut(digest_var) = F::zero();
-        assert!(circuit.check_circuit_satisfiability(&[]).is_err());
+        assert!(circuit.check_circuit_satisfiability(&public_input).is_err());
         *circuit.witness_mut(digest_var) = tmp;
 
         // bad path: bad threshold
         *circuit.witness_mut(threshold_var) = F::from(7u8);
-        assert!(circuit.check_circuit_satisfiability(&[]).is_err());
+        assert!(circuit.check_circuit_satisfiability(&public_input).is_err());
 
         // check input parameter errors
         assert!(circuit
-            .check_aggregate_vk::<E, P>(&vk_vars[..], &bitvec_vars[1..], &agg_vk_var, d_ecc)
+            .check_aggregate_vk::<E, EmulatedSWPointVariable<E>>(
+                &vk_vars[..],
+                &selector_vars[1..],
+                &agg_vk_var,
+                a_ecc
+            )
             .is_err());
         assert!(circuit
             .check_stake_table_digest(&vk_vars[..], &stake_amt_vars[1..], digest_var)
             .is_err());
         assert!(circuit
-            .check_threshold(&stake_amt_vars[..], &bitvec_vars[1..], threshold_var)
+            .check_threshold(&stake_amt_vars[..], &selector_vars[1..], threshold_var)
             .is_err());
 
         Ok(())
     }
 
-    // TODO(binyi): make the API public
-    fn compute_stake_table_hash<F: RescueParameter, E: EmulationConfig<F>>(
-        stake_amts: &[F],
-        vk_points: &[Point<E>],
-    ) -> F {
-        let mut input_vec = vec![];
-        for (&amt, point) in stake_amts.iter().zip(vk_points.iter()) {
-            let x_vec = from_emulated_field(point.get_x());
-            let y_vec = from_emulated_field(point.get_y());
-            input_vec.extend(x_vec);
-            input_vec.extend(y_vec);
-            input_vec.push(amt);
-        }
-        RescueCRHF::sponge_with_bit_padding(&input_vec[..], 1)[0]
+    #[test]
+    fn test_vk_aggregate_te_circuit() -> Result<(), CircuitError> {
+        let d_ecc : Fq377 = MontFp!("122268283598675559488486339158635529096981886914877139579534153582033676785385790730042363341236035746924960903179");
+        test_vk_aggregate_te_circuit_helper::<Fq377, Fr254, Param377>(d_ecc)
```

Comment:
Same here, should we replace Fr254 with Fr377?

Reply:
Same as above